examples/admanager_axis/src/main/java/admanager/axis/v202005/placementservice/CreatePlacements.java

```diff
@@ -0,0 +1,232 @@
+import com.google.api.ads.common.lib.exception.ValidationException;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.common.base.Joiner;
+import com.google.common.collect.Sets;
+import java.rmi.RemoteException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+
+/**
+ * This example creates new placements for various ad unit sizes. To determine which placements
+ * exist, run GetAllPlacements.java.
+ *
+ * <p>Credentials and properties in {@code fromFile()} are pulled from the "ads.properties" file.
+ * See README for more info.
+ */
+public class CreatePlacements {
+
+  /**
+   * Runs the example.
+   *
+   * @param adManagerServices the services factory.
+   * @param session the session.
+   * @throws ApiException if the API request failed with one or more service errors.
+   * @throws RemoteException if the API request failed due to other errors.
+   */
+  public static void runExample(AdManagerServices adManagerServices, AdManagerSession session)
+      throws RemoteException {
+    // Get the PlacementService.
+    PlacementServiceInterface placementService =
+        adManagerServices.get(session, PlacementServiceInterface.class);
+
+    // Get all ad units.
+    List<AdUnit> adUnits = getAllAdUnits(adManagerServices, session);
+
+    // Partition ad units by their size.
+    Set<String> mediumSquareAdUnitIds = Sets.newHashSet();
+    Set<String> skyscraperAdUnitIds = Sets.newHashSet();
+    Set<String> bannerAdUnitIds = Sets.newHashSet();
+
+    for (AdUnit adUnit : adUnits) {
+      if (adUnit.getParentId() != null && adUnit.getAdUnitSizes() != null) {
+        for (AdUnitSize adUnitSize : adUnit.getAdUnitSizes()) {
+          Size size = adUnitSize.getSize();
+          if (size.getWidth() == 300 && size.getHeight() == 250) {
+            mediumSquareAdUnitIds.add(adUnit.getId());
+          } else if (size.getWidth() == 120 && size.getHeight() == 600) {
+            skyscraperAdUnitIds.add(adUnit.getId());
+          } else if (size.getWidth() == 468 && size.getHeight() == 60) {
+            bannerAdUnitIds.add(adUnit.getId());
+          }
+        }
+      }
+    }
+
+    List<Placement> placementsToCreate = new ArrayList<>();
+
+    // Only create placements with one or more ad unit.
+    if (!mediumSquareAdUnitIds.isEmpty()) {
+      // Create medium square placement.
+      Placement mediumSquareAdUnitPlacement = new Placement();
+      mediumSquareAdUnitPlacement.setName(
+          "Medium Square AdUnit Placement #" + new Random().nextInt(Integer.MAX_VALUE));
+      mediumSquareAdUnitPlacement.setDescription(
+          "Contains ad units that can hold creatives of size 300x250");
+      mediumSquareAdUnitPlacement.setTargetedAdUnitIds(
+          mediumSquareAdUnitIds.toArray(new String[] {}));
+
+      placementsToCreate.add(mediumSquareAdUnitPlacement);
+    }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 95:106, 108:117, 119:128.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.